ext/spl/spl_fixedarray.c

```diff
@@ -311,13 +311,13 @@ static zend_object *spl_fixedarray_object_clone(zend_object *old_object)
 
 static zend_long spl_offset_convert_to_long(zval *offset) /* {{{ */
 {
-	zend_ulong idx;
+	zend_long index;
 
 	try_again:
 	switch (Z_TYPE_P(offset)) {
 		case IS_STRING:
-			if (ZEND_HANDLE_NUMERIC(Z_STR_P(offset), idx)) {
-				return idx;
+			if (IS_LONG == is_numeric_str_function(Z_STR_P(offset), &index, NULL)) {
```

Comment:
Why this change? It makes it less strict...

Reply:
Because that's what my brain thought was a good idea at 2AM, but logic was, well if we only accept numeric strings, the logic should probably be similar to how we treat string offsets.

But will revert.